src/CommunityPatch/Patches/Perks/Endurance/Riding/NomadicTraditionsPatch.cs

```diff
@@ -0,0 +1,94 @@
+        0x9C, 0xDC, 0x50, 0x6E, 0xA6, 0x7E, 0xEB, 0xEF
+      }
+    };
+
+    public override void Reset() {}
+
+    public override bool? IsApplicable(Game game)
+    {
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo))
+        return false;
+
+      var hash = TargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes);
+    }
+
+    public override void Apply(Game game) {
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo,
+        postfix: new HarmonyMethod(PatchMethodInfo));
+      Applied = true;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static void Postfix(DefaultPartySpeedCalculatingModel __instance, ref StatExplainer explanation, ref float __result, MobileParty mobileParty, ref int additionalTroopOnFootCount, ref int additionalTroopOnHorseCount) {
+      var hero = mobileParty.LeaderHero;
+      if (hero != null && hero.GetPerkValue(DefaultPerks.Riding.NomadicTraditions)) {
+        //have to recalculate cavalry and mounted footman bonuses, so repeat the code
+        var party = mobileParty.Party;
+        var availableHorses = mobileParty.ItemRoster.NumberOfMounts;
+        var totalTroops = mobileParty.MemberRoster.TotalManCount + additionalTroopOnFootCount + additionalTroopOnHorseCount;
+        var mountedTroops = party.NumberOfMenWithHorse + additionalTroopOnHorseCount;
+        var troopsOnFoot = party.NumberOfMenWithoutHorse + additionalTroopOnFootCount;
+        if (mobileParty.AttachedParties.Count != 0) {
+          foreach (var mobileParty2 in mobileParty.AttachedParties) {
+            totalTroops += mobileParty2.MemberRoster.TotalManCount;
+            mountedTroops += mobileParty2.Party.NumberOfMenWithHorse;
+            troopsOnFoot += mobileParty2.Party.NumberOfMenWithoutHorse;
+            availableHorses += mobileParty2.ItemRoster.NumberOfMounts;
+          }
+        }
+
+        int mountedFootman = Math.Min(availableHorses, troopsOnFoot);
+        var mountedFootmenRatioMethod = typeof(DefaultPartySpeedCalculatingModel).GetMethod("GetMountedFootmenRatioModifier", BindingFlags.Instance | BindingFlags.NonPublic);
+        var defaultMountedFootmanRatio = (float)mountedFootmenRatioMethod.Invoke(__instance, new object[] {totalTroops, mountedFootman});
+        //default Mounted Footman constant is 0.3, so just apply it again
+        var perkMountedFootmanRatio = defaultMountedFootmanRatio;
+        
+        var cavalryRatioMethod = typeof(DefaultPartySpeedCalculatingModel).GetMethod("GetCavalryRatioModifier", BindingFlags.Instance | BindingFlags.NonPublic);
+        var defaultCavalryRatio = (float)cavalryRatioMethod.Invoke(__instance, new object[] {totalTroops, mountedTroops});
+        //default Cavalry constant is 0.6, so do math
+        var perkCavalryRatio = Convert.ToSingle(0.3 * defaultCavalryRatio / 0.6);
```

Comment:
The maths feels wrong here,  I don't think we should be dividing by 0.6 here. I'd expect that cavalry constant to go to 0.78f not 0.90f

Reply:
I'm multiplying by 0.6 to get the flat cavalry ratio (see `GetCavalryRatioModifier` on the same class. Although at this point I should just have done the division manually)
yeah so both for both of these, I interpreted the perk as "gives 30% bonus on the flat ratio". So all said and done, with the perk active, total cavalry bonus would be `(0.6 + 0.3) * cavalryRatio` and mounted footman `(0.3 + 0.3) * mountedFootmanRatio`.
What you're suggesting is that I apply the 30% bonus AFTER the initial cut, so perk ratio is `0.3 * (0.6 + cavalryRatio)`? How did you get to 0.78f?

Reply:
I'm thinking the 'constant' multiplier should be multiplied by 1.3, i.e. 0.6*1.3 = 0.78f  for the 30% bonus. but this is kind of semantics. so i'm happy for it to go either way.